07-practical-work-1/COURSE_MATERIAL.md

```diff
@@ -27,44 +27,49 @@ This work is licensed under the [CC BY-SA 4.0][license] license.
 - [Objectives](#objectives)
 - [Group composition](#group-composition)
 - [Grading criteria](#grading-criteria)
-  - [Category 1 - Git, GitHub and Markdown](#category-1---git-github-and-markdown)
-  - [Category 2 - Java, IntelliJ IDEA and Maven](#category-2---java-intellij-idea-and-maven)
-  - [Category 3 - Java IOs](#category-3---java-ios)
-  - [Category 4 - Presentation and questions](#category-4---presentation-and-questions)
+  - [Category 1 - Meta](#category-1---meta)
+  - [Category 2 - Git, GitHub and Markdown](#category-2---git-github-and-markdown)
+  - [Category 3 - Java, IntelliJ IDEA and Maven](#category-3---java-intellij-idea-and-maven)
+  - [Category 4 - Java IOs](#category-4---java-ios)
+  - [Category 5 - Presentation and questions](#category-5---presentation-and-questions)
 - [Constraints](#constraints)
 - [Remarks](#remarks)
 - [Submission](#submission)
 - [Grades and feedback](#grades-and-feedback)
 - [Finished? Was it easy? Was it hard?](#finished-was-it-easy-was-it-hard)
+- [Sources](#sources)
 
 ## Introduction
 
-We use CLI tools every day. For example, we use `git` to manage our code, `mvn`
-to build our projects, `java` to run our programs, etc.
+We use command line interface (CLI) tools every day. For example, we use Git to
+manage our code (with its `git` command line interface), Maven to build our
+projects (with its `mvn` and `mvnw` command line interface), Java (with its
+`javac` and `java` command line interface) to run our programs, etc.
 
 In this practical work, you will create a CLI using
 [picocli](https://picocli.info/) to process files.
 
 The CLI will take an input file and an output file as arguments. It will also
-take optional arguments such as the input file encoding and the output file
-encoding (the default will be UTF-8 for example). The CLI will process the input
-file and write the result in the output file. It will display a message on
-success and a message on failure.
+take optional arguments to customize the CLI (such as the input file encoding
+and the output file encoding - the default being UTF-8 for example). The CLI
+will process the input file and write the result in the output file. It will
+display a message on success and a message on failure.
 
 You have the freedom to define what the CLI will do. You can be creative! For
-example, you can choose to transform the the input text file to the output text
-file (uppercase/lowercase/find/replace/count number of words/etc.), to convert a
-PNG to JPEG, add metadata to existing TIFF files, etc. If you do not have any
-idea, come to see us and we can give you some ideas.
+example, you can choose to transform a text file (find/replace/count number of
+occurrences/etc.), to grayscale a JPEG/PNG binary file manipulating its pixel
+values, add metadata to existing TIFF files, etc.
```

Comment:
If my memory is right, we have decided to allow external libraries as long as the opening/closing part is managed by the students with the Java classes seen during the course.

This would allow students to have the flexibility to use these external libs but they must manually manage the files instead of delegating this part entirely to external libs.

What do you think? Both solutions are fine by me.

Reply:
The solution you suggest is perfect for me